src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,259 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from multiprocessing import Queue
+import threading
+
+from avocado.core.exceptions import TestFail
+from dmg_utils import get_storage_query_device_info, get_dmg_response
+from exception_utils import CommandFailure
+from ior_utils import run_ior, thread_run_ior
+from job_manager_utils import get_job_manager
+from led import VmdLedStatus
+from nvme_utils import set_device_faulty
+
+
+class NvmeFaultReintegrate(VmdLedStatus):
+    """
+    Test Class Description: This test runs
+    NVME fault and reintegration test cases.
+
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-ancestors
+    def setUp(self):
+        """Set up for test case."""
+        super().setUp()
+        self.daos_command = self.get_daos_command()
+
+    def verify_dev_led_state(self, device, dev_state="NORMAL", led_state="OFF"):
+        """Verify device dev_state and led_state.
+
+        Args:
+            device (str): the device description.
+            dev_state (str): expect dev_state, default to "NORMAL".
+            led_state (str): expect led_state, default to "OFF".
+
+        Returns:
+            bool: True if the expected dev_state and led_state.
+
+        """
+        return self.check_result(self.get_led_status_value(device), dev_state, led_state)
```

Comment:
For simplification, `self.get_led_status_value(device)` is the same as `get_dmg_response(self, self.dmg.storage_led_check, ids=device)` and is there anything left for this test to inherit from VmdLedStatus?
```suggestion
        return self.check_result(get_dmg_response(
            self, self.dmg.storage_led_check, ids=device), dev_state, led_state)
```

Reply:
The response of the 2 methods are different:  get_dmg_response do not has key "response", and get_led_status_value is under key "response":

(1)res = get_dmg_response(self, self.dmg.storage_led_check, ids=device)
{'host_errors': {},  
 'host_storage_map': {'4176933689015108273': {'storage': {'nvme_devices': None, 'scm_modules': None, 'scm_namespaces': None, 'scm_mount_points': None, 'smd_info': {'devices': [{'uuid': '', 'tgt_ids': None, 'dev_state': 'NORMAL', 'led_state': 'OFF', 'rank': 0, 'total_bytes': 0, 'avail_bytes': 0, 'cluster_size': 0, 'health': None, 'tr_addr': '2a0005:87:00.0'}], 'pools': None}, 'reboot_required': False, 'mem_info': {'hugepage_size_kb': 0, 'mem_total': 0, 'mem_free': 0, 'mem_available': 0}}, 'hosts': 'wolf-217:10001'}}}

(2)response = get_led_status_value(device)
{'response': {
 'host_errors': {}, 
 'host_storage_map': {'4176933689015108273': {'storage': {'nvme_devices': None, 'scm_modules': None, 'scm_namespaces': None, 'scm_mount_points': None, 'smd_info': {'devices': [{'uuid': '', 'tgt_ids': None, 'dev_state': 'NORMAL', 'led_state': 'OFF', 'rank': 0, 'total_bytes': 0, 'avail_bytes': 0, 'cluster_size': 0, 'health': None, 'tr_addr': '2a0005:87:00.0'}], 'pools': None}, 'reboot_required': False, 'mem_info': {'hugepage_size_kb': 0, 'mem_total': 0, 'mem_free': 0, 'mem_available': 0}}, 'hosts': 'wolf-217:10001'}}}, 'error': None, 'status': 0}